tests/approaches/test_grammar_search_invention_approach.py

```diff
@@ -14,7 +14,8 @@
     _RelaxationHeuristicEnergyBasedScoreFunction, _TaskPlanningScoreFunction,
     _ExactHeuristicEnergyBasedScoreFunction,
     _RelaxationHeuristicCountBasedScoreFunction,
-    _ExactHeuristicCountBasedScoreFunction, _BranchingFactorScoreFunction)
```

Comment:
can you make sure that your tests aren't too slow? local should take ~60 seconds total and you can also just try `pytest -s tests/approaches/test_grammar_search_invention_approach.py --durations=0` to see the full speed printout for this file

Reply:
looks okay, here's the printout:
```
================================================================== slowest durations ==================================================================
11.09s call     tests/approaches/test_grammar_search_invention_approach.py::test_relaxation_energy_score_function
9.59s call     tests/approaches/test_grammar_search_invention_approach.py::test_prediction_error_score_function
5.22s call     tests/approaches/test_grammar_search_invention_approach.py::test_hadd_match_score_function
5.14s call     tests/approaches/test_grammar_search_invention_approach.py::test_exact_energy_score_function
2.66s call     tests/approaches/test_grammar_search_invention_approach.py::test_expected_nodes_score_function
2.03s call     tests/approaches/test_grammar_search_invention_approach.py::test_task_planning_score_function
1.10s call     tests/approaches/test_grammar_search_invention_approach.py::test_count_score_functions
0.82s call     tests/approaches/test_grammar_search_invention_approach.py::test_branching_factor_score_function
0.06s call     tests/approaches/test_grammar_search_invention_approach.py::test_predicate_grammar
0.02s call     tests/approaches/test_grammar_search_invention_approach.py::test_predicate_search_heuristic_base_classes
```